src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesProvisioningLimits.java

```diff
@@ -0,0 +1,133 @@
+package org.csanchez.jenkins.plugins.kubernetes;
+
+import com.google.common.annotations.VisibleForTesting;
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.model.Node;
+import hudson.model.Queue;
+import jenkins.metrics.api.Metrics;
+import jenkins.model.Jenkins;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Implements provisioning limits for clouds and pod templates
+ */
+@Extension
+public final class KubernetesProvisioningLimits {
+    private static final Logger LOGGER = Logger.getLogger(KubernetesProvisioningLimits.class.getName());
+
+    /**
+     * Tracks current number of kubernetes agents per pod template
+     */
+    private final transient Map<String, AtomicInteger> podTemplateCounts = Collections.synchronizedMap(new HashMap<>());
```

Comment:
```suggestion
    private final Map<String, AtomicInteger> podTemplateCounts = Collections.synchronizedMap(new HashMap<>());
```

since presumably you were not trying to serialize it to begin with